src/main/scala/subsystem/RTC.scala

```diff
@@ -12,15 +12,17 @@ trait HasRTCModuleImp extends LazyModuleImp {
   private val rtcFreq = outer.p(DTSTimebase)
   private val internalPeriod: BigInt = pbusFreq / rtcFreq
 
+  val pbus = outer.locateTLBusWrapper(PBUS)
   // check whether pbusFreq >= rtcFreq
   require(internalPeriod > 0)
   // check wehther the integer division is within 5% of the real division
   require((pbusFreq - rtcFreq * internalPeriod) * 100 / pbusFreq <= 5)
 
   // Use the static period to toggle the RTC
-  val (_, int_rtc_tick) = Counter(true.B, internalPeriod.toInt)
-
-  outer.clintOpt.foreach { clint =>
-    clint.module.io.rtcTick := int_rtc_tick
+  chisel3.withClockAndReset(pbus.module.clock, pbus.module.reset) {
```

Comment:
I chose to explicitly look up the PBUS because the calculations above are using the PBus dtsFrequency. 

What i can do is grab the clocks off the clint as you suggest, and then look up the bus to which it is attached to find the appropriate frequency for the above stuff --- or am I just very mistaken that the counter needs to be resident in a domain with the frequency above?

Reply:
Hmm yeah the semantics of this code seem pretty sketchy. On second look I agree that using `PBUS` here isn't any worse than the use of `PeripheryBusKey` above ... just continues the assumption that PBUS and CBUS and CLINT all have the same clock. I want to tell you to do what you said, since I think it would be more consistent, but I'm not especially keen to go play whackamole with every case where the value is currently being Config-ured for PBUS but not CBUS. Let's take this as is and kick the can down the road to https://github.com/chipsalliance/rocket-chip/pull/2643

Reply:
Yeah that make sense.  I think we're interested in having the PBUS run a different frequency immediately, but we can use our own mixin until this is sorted.